File: 1]. DSA/1]. Data Structures/01]. Array/C++/_028)_Maximum_Subarray_Sum.cpp
#include<iostream>
using namespace std;

int maxSubarraySum(int arr[], int n)
{
    int res = arr[0];
    int maxEnding = arr[0];
    for(int i=1; i<n; i++)
    {
        maxEnding = max(maxEnding + arr[i], arr[i]);
        res = max(res, maxEnding);
    }
    return res;
}

int main()
{
    int arr[] = {2, 3, -8, 7, -1, 2, 3}, n=7;
    cout<<"Maximum SubArray Sum = "<<maxSubarraySum(arr,n)<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_028)_Maximum_Subarray_Sum.dart
import 'dart:math';

int maxSubArraySum(List<int> arr) {
  int res = arr[0];
  int maxEnding = arr[0];

  for (int i = 1; i < arr.length; i++) {
    maxEnding = max(maxEnding + arr[i], arr[i]);
    res = max(res, maxEnding);
  }
  return res;
}

void main() {
  var arr = [2, 3, -8, 7, -1, 2, 3];
  print('Maximum sub-array sum is ${maxSubArraySum(arr)}.');
}
